tests/parser/dictionary1.py

```diff
@@ -24,4 +24,6 @@ def test():
           """
         }
 
+    {**self._derived_units, **derived_units}
```

Comment:
I have a doubt @certik,
For the above example, we store `nullptr` for keys
```py
└-Module
  |-body=↧
  | └-Expr
  |   └-value=Dict
  |     |-keys=↧
  |     └-values=↧
  |       |-Attribute
  |       | |-value=Name
  |       | | |-id=self
  |       | | └-expr_contextType=Load
  |       | |-attr=_derived_units
  |       | └-expr_contextType=Load
  |       └-Name
  |         |-id=derived_units
  |         └-expr_contextType=Load
  └-type_ignores=↧
```
But CPython stores `None`
```py
Module(
   body=[
      Expr(
         value=Dict(
            keys=[
               None,
               None],
            values=[
               Attribute(
                  value=Name(id='self', ctx=Load()),
                  attr='_derived_units',
                  ctx=Load()),
               Name(id='derived_units', ctx=Load())]))],
   type_ignores=[])
```
Should we store None as well? `or` the above LPython Output looks good?

Reply:
How do we handle None in other contexts?

Reply:
We only store Constant None in the AST
Example:
```py
x = {None: "none"}
```
CPython
```py
Module(
   body=[
      Assign(
         targets=[
            Name(id='x', ctx=Store())],
         value=Dict(
            keys=[
               Constant(value=None)],
            values=[
               Constant(value='none')]))],
   type_ignores=[])
```

Reply:
I see. It seems to me that `keys` and `values` should always have the same length, like in CPython, so storing ConstantNone seems like a better solution for LPython? Let me know what you think.

Reply:
Now, I get some clarity!
So, here we cannnot have `nullptr` when the key is not available.
Consider an example:
```py
{
    **a,
    x: "Text",
    **y,
    z: "Text"
}
```
```py
└-Module
  |-body=↧
  | └-Expr
  |   └-value=Dict
  |     |-keys=↧
  |     | |-Name
  |     | | |-id=x
  |     | | └-expr_contextType=Load
  |     | └-Name
  |     |   |-id=z
  |     |   └-expr_contextType=Load
  |     └-values=↧
  |       |-Name
  |       | |-id=a
  |       | └-expr_contextType=Load
  |       |-ConstantStr
  |       | |-value="Text"
  |       | └-kind=()
  |       |-Name
  |       | |-id=y
  |       | └-expr_contextType=Load
  |       └-ConstantStr
  |         |-value="Text"
  |         └-kind=()
  └-type_ignores=↧
```
which is wrong, as the index gets wrongly matched.
Now, what should we do with this?
Add `ConstantNone`? I think that would cause the conflict!
Consider another example:
```py
{
    **a,
    None: "Text",
    **y,
    z: "Text"
}
```
```py
Module(
   body=[
      Expr(
         value=Dict(
            keys=[
               None,
               Constant(value=None),
               None,
               Name(id='z', ctx=Load())],
            values=[
               Name(id='a', ctx=Load()),
               Constant(value='Text'),
               Name(id='y', ctx=Load()),
               Constant(value='Text')]))],
   type_ignores=[])
```
This is the simple example in which the key is not available
```py
a = {
    "a": "Text 1"
}
b = {
    "b" : "Text 2"
}

x = {
    **a,
    None: "Text 3",
    **b,
    "c": "Text 4"
}

print(x["a"])
```
I think we can add a new node for this?

Reply:
I would just do whatever CPython does. Do we have means to store None in the dict? If so, then I would store it just like CPython does, because then we don't have to worry about the design. We just do whatever they do, and we can check it against them.

If there is something that our AST cannot fundamentally represent, then let's discuss that and find a solution. Otherwise I would just follow CPython's lead on this.

Reply:
Currently, we only have `ConstatntNone`. So, I'm planning to create a new node `None` and pass this for the dict.